tasks/process_flakes.py

```diff
@@ -61,13 +69,32 @@ def run_impl(
             "Successfully processed flakes",
             extra=dict(repoid=repo_id, commit=commit_id_list),
         )
+
         return {"successful": True}
 
 
-def get_test_instances(commit_id: str, repo_id: int, branch: str) -> list[TestInstance]:
+def get_test_instances(
+    commit_id: str,
+    repo_id: int,
+    branch: str,
+    flaky_tests: list[str],
+) -> list[TestInstance]:
+    # get test instances on this repo commit branch combination that either:
+    # - failed
+    # - belong to an already flaky test
     test_instances = list(
         TestInstance.objects.filter(
-            commitid=commit_id, repoid=repo_id, branch=branch
+            Q(commitid=commit_id)
```

Comment:
[nit]

I think this can be refactored to be easier to read if you break up the conditions and give them a name

```python
basic_filters = Q(commitid=commit_id) & Q(repoid=repo_id) & Q(branch=branch)
test_failed_filters = Q(outcome=TestInstance.Outcome.ERROR.value) | Q(outcome=TestInstance.Outcome.FAILURE.value)
test_already_flaky_filters= Q(outcome=TestInstance.Outcome.PASS.value) & Q(test_id__in=flaky_tests)
test_instances = list( 
  TestInstance.objects.filter( 
    basic_filters & (test_failed_filters | test_already_flaky_filters)
  ).all()
)
```